C/include/c4Observer.h

```diff
@@ -107,18 +107,21 @@ C4API_BEGIN_DECLS
         @param docID  The ID of the document that changed.
         @param sequence  The sequence number of the change.
         @param context  user-defined parameter given when registering the callback. */
-    typedef void (*C4DocumentObserverCallback)(C4DocumentObserver* observer,
-                                               C4String docID,
-                                               C4SequenceNumber sequence,
-                                               void * C4NULLABLE context);
+    typedef void (*C4CollectionDocObserverCallback)(C4DocumentObserver* observer,
+                                                   C4CollectionSpec collection,
```

Comment:
If we expose the collection of C4DocumentObserver, then we don't need to provide collection as argument in this callback. I this is the identity: observer->collection == collection. That is, there is a redundancy here.

Reply:
Why not? A getter of CollectonSpec, which is a pure value, is simple and safe, removing the redundancy.

Reply:
It will require the addition of c4docobs_getCollectionSpec.  A simple getter is not usable for the C API.